common/apps.go

```diff
@@ -1,87 +1,65 @@
 package common
 
 import (
-	"context"
 	"fmt"
+	"time"
 
 	"github.com/giantswarm/apiextensions-application/api/v1alpha1"
+	"github.com/giantswarm/clustertest/pkg/logger"
 	. "github.com/onsi/ginkgo/v2"
 	. "github.com/onsi/gomega"
-	"github.com/onsi/gomega/format"
-	"github.com/onsi/gomega/types"
 	ctrl "sigs.k8s.io/controller-runtime/pkg/client"
 
 	"github.com/giantswarm/cluster-test-suites/internal/state"
 )
 
+const deployedStatus = "deployed"
+
 func runApps() {
 	Context("default apps", func() {
-		var customFormatterKey format.CustomFormatterKey
-		BeforeEach(func() {
-			customFormatterKey = format.RegisterCustomFormatter(func(value interface{}) (string, bool) {
-				app, ok := value.(v1alpha1.App)
-				if ok {
-					return fmt.Sprintf("App: %s/%s, Status: %s, Reason: %s", app.Namespace, app.Name, app.Status.Release.Status, app.Status.Release.Reason), true
-				}
-
-				return "", false
-			})
-		})
-
 		It("all default apps are deployed without issues", func() {
-			ctx := context.Background()
 
 			// We need to wait for default-apps to be deployed before we can check all apps.
 			defaultAppsAppName := fmt.Sprintf("%s-%s", state.GetCluster().Name, "default-apps")
-			Eventually(state.GetFramework().GetApp, "30s").WithContext(ctx).WithArguments(defaultAppsAppName, state.GetCluster().Organization.GetNamespace()).Should(HaveAppStatus("deployed"))
+			Eventually(func() error {
+				app, err := state.GetFramework().GetApp(state.GetContext(), defaultAppsAppName, state.GetCluster().Organization.GetNamespace())
+				if err != nil {
+					return err
+				}
+				return checkAppStatus(app)
+			}).
+				WithTimeout(30 * time.Second).
+				WithPolling(50 * time.Millisecond).
+				Should(Succeed())
+
+			// Wait for all default-apps apps to be deployed
+			Eventually(func() error {
+				appList := &v1alpha1.AppList{}
+				err := state.GetFramework().MC().List(state.GetContext(), appList, ctrl.InNamespace(state.GetCluster().Organization.GetNamespace()), ctrl.MatchingLabels{"giantswarm.io/managed-by": defaultAppsAppName})
+				if err != nil {
+					return err
+				}
 
-			managementClusterKubeClient := state.GetFramework().MC()
-			appList := &v1alpha1.AppList{}
-			err := managementClusterKubeClient.List(ctx, appList, ctrl.InNamespace(state.GetCluster().Organization.GetNamespace()), ctrl.MatchingLabels{"giantswarm.io/managed-by": defaultAppsAppName})
-			Expect(err).ShouldNot(HaveOccurred())
+				for _, app := range appList.Items {
```

Comment:
Should we collect all non-nil errors? Maybe cluster-api's `kerrors.NewAggregate(listOfErrors)` kind of makes sense, or we make it really simple and concatenate all errors with a newline.

This is really optional since below, I can see that the app status of all other apps is logged anyhow. So feel free to merge without this minor detail.

Reply:
I did wonder that. I wasn't sure which way to go. 

I didn't want to potentially flood the logs with everything as it could get noisy easily. With the approach, it'll eventually work it's way through the still pending until they're all running. The only thing it wont do it tell us if we've timed out with more than one still pending.

Reply:
Yeah, I prefer having the more info:
```
Common tests default apps all default apps are deployed without issues
/app/common/apps.go:21
  {"level":"info","ts":"2023-08-29T07:28:49Z","msg":"App t-v31jfgr-aws-pod-identity-webhook status is currently ''"}
  {"level":"info","ts":"2023-08-29T07:28:49Z","msg":"App t-v31jfgr-cert-manager status is currently 'invalid-manifest'"}
  {"level":"info","ts":"2023-08-29T07:28:49Z","msg":"App t-v31jfgr-external-dns status is currently 'failed'"}
  {"level":"info","ts":"2023-08-29T07:28:49Z","msg":"App t-v31jfgr-vertical-pod-autoscaler status is currently ''"}
  {"level":"info","ts":"2023-08-29T07:28:59Z","msg":"App t-v31jfgr-aws-pod-identity-webhook status is currently ''"}
  {"level":"info","ts":"2023-08-29T07:28:59Z","msg":"App t-v31jfgr-cert-manager status is currently 'invalid-manifest'"}
  {"level":"info","ts":"2023-08-29T07:28:59Z","msg":"App t-v31jfgr-external-dns status is currently 'failed'"}
  {"level":"info","ts":"2023-08-29T07:28:59Z","msg":"App t-v31jfgr-vertical-pod-autoscaler status is currently ''"}
  {"level":"info","ts":"2023-08-29T07:29:09Z","msg":"App t-v31jfgr-aws-pod-identity-webhook status is currently ''"}
  {"level":"info","ts":"2023-08-29T07:29:09Z","msg":"App t-v31jfgr-cert-manager status is currently 'invalid-manifest'"}
  {"level":"info","ts":"2023-08-29T07:29:09Z","msg":"App t-v31jfgr-external-dns status is currently 'failed'"}
  {"level":"info","ts":"2023-08-29T07:29:09Z","msg":"App t-v31jfgr-vertical-pod-autoscaler status is currently ''"}
  {"level":"info","ts":"2023-08-29T07:29:19Z","msg":"App t-v31jfgr-aws-pod-identity-webhook status is currently ''"}
  {"level":"info","ts":"2023-08-29T07:29:19Z","msg":"App t-v31jfgr-cert-manager status is currently 'invalid-manifest'"}
  {"level":"info","ts":"2023-08-29T07:29:19Z","msg":"App t-v31jfgr-external-dns status is currently 'failed'"}
  {"level":"info","ts":"2023-08-29T07:29:19Z","msg":"App t-v31jfgr-vertical-pod-autoscaler status is currently ''"}
  {"level":"info","ts":"2023-08-29T07:29:29Z","msg":"App t-v31jfgr-aws-pod-identity-webhook status is currently ''"}
  {"level":"info","ts":"2023-08-29T07:29:29Z","msg":"App t-v31jfgr-cert-manager status is currently 'invalid-manifest'"}
  {"level":"info","ts":"2023-08-29T07:29:29Z","msg":"App t-v31jfgr-external-dns status is currently 'failed'"}
  {"level":"info","ts":"2023-08-29T07:29:29Z","msg":"App t-v31jfgr-vertical-pod-autoscaler status is currently ''"}
  {"level":"info","ts":"2023-08-29T07:29:39Z","msg":"App t-v31jfgr-aws-pod-identity-webhook status is currently ''"}
  {"level":"info","ts":"2023-08-29T07:29:39Z","msg":"App t-v31jfgr-cert-manager status is currently 'invalid-manifest'"}
  {"level":"info","ts":"2023-08-29T07:29:39Z","msg":"App t-v31jfgr-external-dns status is currently 'failed'"}
  {"level":"info","ts":"2023-08-29T07:29:39Z","msg":"App t-v31jfgr-vertical-pod-autoscaler status is currently ''"}
  {"level":"info","ts":"2023-08-29T07:29:49Z","msg":"App t-v31jfgr-aws-pod-identity-webhook status is currently ''"}
  {"level":"info","ts":"2023-08-29T07:29:49Z","msg":"App t-v31jfgr-cert-manager status is currently 'invalid-manifest'"}
  {"level":"info","ts":"2023-08-29T07:29:49Z","msg":"App t-v31jfgr-external-dns status is currently 'failed'"}
  {"level":"info","ts":"2023-08-29T07:29:49Z","msg":"App t-v31jfgr-vertical-pod-autoscaler status is currently ''"}
  {"level":"info","ts":"2023-08-29T07:29:59Z","msg":"App t-v31jfgr-aws-pod-identity-webhook status is currently ''"}
  {"level":"info","ts":"2023-08-29T07:29:59Z","msg":"App t-v31jfgr-cert-manager status is currently 'invalid-manifest'"}
  {"level":"info","ts":"2023-08-29T07:29:59Z","msg":"App t-v31jfgr-external-dns status is currently 'failed'"}
  {"level":"info","ts":"2023-08-29T07:29:59Z","msg":"App t-v31jfgr-vertical-pod-autoscaler status is currently ''"}
  {"level":"info","ts":"2023-08-29T07:30:09Z","msg":"App t-v31jfgr-aws-pod-identity-webhook status is currently ''"}
```

Reply:
Just added a slight update to also include a log line at the start of each check indicating how many apps are being checked.